Convex_hull_3/include/CGAL/Convex_hull_3/intersections.h

```diff
@@ -0,0 +1,706 @@
+// Copyright (c) 2022 INRIA (France) and GeometryFactory (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0-or-later OR LicenseRef-Commercial
+//
+//
+// Author(s)     : Samuel Hornus, Sébastien Loriot and Léo Valque
+//
+
+#ifndef CGAL_CONVEX_HULL_3_INTERSECTIONS_H
+#define CGAL_CONVEX_HULL_3_INTERSECTIONS_H
+
+#include <CGAL/license/Convex_hull_3.h>
+
+#include <CGAL/Named_function_parameters.h>
+#include <CGAL/boost/graph/named_params_helper.h>
+
+#include <CGAL/Kernel_23/internal/Has_boolean_tags.h>
+
+#include <CGAL/Convex_hull_hierarchy.h>
+#include <CGAL/intersections.h>
+
+#include <CGAL/IO/helpers.h>
+
+#include <CGAL/Container_helper.h>
+#include <CGAL/Dynamic_property_map.h>
+#include <CGAL/property_map.h>
+
+#include <boost/range/value_type.hpp>
+#include <boost/range/reference.hpp>
+
+#include <boost/graph/adjacency_list.hpp>
+
+#include <CGAL/Real_timer.h>
+
+#include <vector>
+
+namespace CGAL {
+
+template <class T, template <class...> class Template>
+inline constexpr bool is_instance_of_v = false;
+
+template <template <class...> class Template, class... Args>
+inline constexpr bool is_instance_of_v<Template<Args...>, Template> = true;
+
+namespace Convex_hull_3 {
+
+namespace predicates_impl{
+
+// template class to deduce the GT from Convex and NamedParameters
+template<class Convex,
```

Comment:
if this part is the same as CH3, it should be put in a helper file and included by both files. Btw, Sven recently updated it in `main`.